### PR TITLE
Buffer batch for async producers

### LINKED ISSUE #214: Batch producer with transformed streams
Hi,
I've done a wrapper for the producer which buffer a configurable amount of messages before sending them to kafka. The messages are sent either when a timeout triggers or a max amount of messages are sent.
Would it be interesting to create a pull request in order to integrate it to this project?
Other kafka client libs such us the Java one have already integrated this functionality, and the performance is really much better.
Cheers

### PR DESCRIPTION
**Pull request by carlessistare**
Other kafka connectors such as the Java one have the possibility to configurate an optional batch (client-side) for async producer mode, in order to send batches of messages instead of single messages at time. This increases performance because the kafka server doesnt need to answer ACK at every message.

The options are:
- Batch size in bytes
- Batch age in miliseconds

This will resolve #214

### CODE REVIEW

**Comment by mazerte** [comment]
+1

**Comment by haio** [comment]
@carlessistare could you also update the document for this change.

**Comment by carlessistare** [comment]
Sorry @haio I'm not sure of which document you want me to update, could you precise? I'll do it with pleasure though

**Comment by carlessistare** [comment]
@haio I just added some doc on README.md file
c9815987b6f93cbb8a8fab93314f4fba1a34ca28

File: README.md [hunk 1]
```diff
@@ -16,10 +16,11 @@ Follow the [instructions](http://kafka.apache.org/documentation.html#quickstart)
 
 # API
 ## Client
-### Client(connectionString, clientId, [zkOptions])
+### Client(connectionString, clientId, [zkOptions], [noAckBatchOptions])
 * `connectionString`: Zookeeper connection string, default `localhost:2181/`
 * `clientId`: This is a user-supplied identifier for the client application, default `kafka-node-client`
 * `zkOptions`: **Object**, Zookeeper options, see [node-zookeeper-client](https://github.com/alexguan/node-zookeeper-client#client-createclientconnectionstring-options)
+* `noAckBatchOptions`: **Object**, when requireAcks is disabled on Producer side we can define the batch properties, 'noAckBatchSize' in bytes and 'noAckBatchAge' in miliseconds. The default value is `{ noAckBatchSize: 500, noAckBatchAge: 300 }`
```
Comment (macobo) [inline README.md]: typo: `in milliseconds`

**Pull request merged**